instructor/common.py

```diff
@@ -195,6 +195,18 @@ def assignments_edit(request, assignment_pk):
 
     return redirect('common_assignment_modify')
 
+
+def check_assignment_owner(func):
+    def check_owner(request):
+        assignment_id = request.session['assignment_id']
+        assignment = models.Assignment.objects.get(id=assignment_id)
+        user = get_object_or_404(User, username=request.user)
```

Comment:
You're right, I guess Django is converting `User` to a `str` before doing the query, and since `str(user)` is the username this works.